core/src/test/scala/kanaloa/reactive/dispatcher/queue/QueueProcessorSpec.scala

```diff
@@ -107,13 +108,37 @@ class QueueProcessorSpec extends SpecWithActorSystem with Eventually with Backen
         }
       }
 
-    "attempt to keep the number of Workers at the minimumWorkers" in withQueueProcessor() { (qp, queueProbe, metricsCollector, testBackend, workerFactory) ⇒
-      //current workers are 5, minimum workers are 3, so killing 4 should result in 2 new recreate attempts
-      workerFactory.probeMap.keys.take(4).foreach(workerFactory.killAndRemoveWorker)
+    "attempt to keep the number of Workers at the minimumWorkers when worker dies" in withQueueProcessor(ProcessingWorkerPoolSettings(healthCheckInterval = 10.milliseconds)) {
+      (qp, queueProbe, metricsCollector, testBackend, workerFactory) ⇒
+        //current workers are 5, minimum workers are 3, so killing 4 should result in 2 new recreate attempts
+        workerFactory.probeMap.keys.take(4).foreach(workerFactory.killAndRemoveWorker)
+        eventually {
+          qp.underlyingActor.workerPool should have size 3
+          testBackend.timesInvoked shouldBe 7 //2 new invocations should have happened
+          workerFactory.probeMap should have size 3 //should only be 3 workers
+        }
+    }
+
+    "attempt to retry create Workers until it hits the minimumWorkers" in {
+      val settings = ProcessingWorkerPoolSettings(minPoolSize = 2, startingPoolSize = 2, healthCheckInterval = 10.milliseconds)
+
+      val testBackend = new Backend {
```

Comment:
This test Backend would be another example of where creating that Backend where you control failures/passes could be reused.